java-frontend/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -270,10 +297,31 @@ public void visitMethodInvocation(MethodInvocationTree syntaxNode) {
           constraintManager.setValueFactory(new WrappedValueFactory(value));
         }
       }
+
+      if (methodCanNotBeOverriden(symbol)) {
+        // we rely on X-Procedural analysis
+        return;
+      }
+
       // close any resource used as argument, even for unknown methods
       closeArguments(syntaxNode.arguments(), 1);
     }
 
+    private static boolean methodCanNotBeOverriden(Symbol symbol) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "methodCanNotBeOverriden" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)